### PR TITLE
multi-field placeholders

### LINKED ISSUE #490: placeholders when multiple fields are empty
Right now, a placeholder for a group of fields can specify which field to check, e.g.

```yaml
_groups:
  group1:
    fields:
      - field1
      - field2
    _placeholder:
      ifEmpty: field1
```

We should allow people to specify more than one field in `ifEmpty`, since there are frequently fields that should display placeholders if both need data (for example, the url _and_ text of a link).

Specifying these options in an array doesn't allow us to define which logical operator we want, and some larger config object is burdensome, so I propose we use a simple, case-insensitive logical string:

```yaml
ifEmpty: field1 and field2
ifEmpty: field1 AND field2
# would display if both field1 and field2 are empty

ifEmpty: field1 or field2
ifEmpty: field1 OR field2
# would display if either field1 or field2 are empty
```

As a special _one more thing_, I'd like to also add support for my favorite logical operator, XOR:

```yaml
ifEmpty: field1 xor field2
ifEmpty: field1 XOR field2
# would display if only field1 or field2 are empty
```

### PR DESCRIPTION
**Pull request by yoshokatana**
Allow placeholders to check **TWO FIELDS** rather than just one. Great for groups, especially link `url` + `text` pairs. Use natural language to specify which properties to check.

```
field1 AND field2
field1 or field2
field1 XOR field2
```

Fixes #490

_Note:_ We're not allowing an arbitrary number of fields because that would require writing a [recursive descent parser](http://blog.oleganza.com/post/106246432/recursive-descent-parser-in-javascript).

### CODE REVIEW

File: behaviors/README.md [hunk 1]
```diff
@@ -231,6 +231,18 @@ _groups:
       ifEmpty: title
 ```
 
+Placeholders may check two different fields with a logical operator. This is handy for components with editable links, as you'll usually want to display a placeholder when _either_ the url or the link text are empty. Operators are case-insensitive, and you can use `AND`, `OR`, or `XOR`.
```
Comment (davidandrewfrey) [inline behaviors/README.md]: Is this functionality limited to 2 fields or is it 2 or more with logical operators between each pair?

Comment (yoshokatana) [inline behaviors/README.md]: limited to two fields

File: decorators/placeholder.js [hunk 2]
```diff
@@ -112,17 +112,68 @@ function isFieldEmpty(data) {
...
+  tokens = ifEmpty.split(' ');
+
+  if (tokens.length === 1) {
+    // check a single field to see if it's empty
+    return isFieldAndEmpty(tokens[0], data);
+  } else if (tokens.length === 3) {
+    // check multiple fields
+    // note: for now this is limited to checking two fields
+    // e.g. foo AND bar, foo OR bar, foo XOR bar
+    return compareMultipleFields(tokens, data);
```
Comment (davidandrewfrey) [inline decorators/placeholder.js]: Maybe rename this function `compareTwoFields` for now until it's comparing 2 or more?

Comment (yoshokatana) [inline decorators/placeholder.js]: sounds good

**Comment by davidandrewfrey** [comment]
⭐ ⭐ ⭐ ⭐ ¾

**Pull request merged**